### PR TITLE
Json improvment

### PR DESCRIPTION
**Pull request by ppolakiewicz**
In Angular app I can create interface that map json object and it dosen't look nice when I have to use uppercase fields so I changed them to lowercase and moved do enums.

Also the game list or game history was encapsulated by object.
There is not need for that, I changed it to list so Angular could handle them as stream.

### CODE REVIEW

File: enumeration.py [hunk 1]
```diff
@@ -39,3 +39,42 @@ class RoundAdvantage(IntEnum):
     TIME = 0
     BOT_1 = 1
     BOT_2 = 2
+
+
+@unique
+class GamesListFileField(Enum):
```
Comment (devdo-eu) [inline enumeration.py:45]: `GameListFileFields` will be more accurate

File: enumeration.py [hunk 2]
```diff
@@ -39,3 +39,42 @@ class RoundAdvantage(IntEnum):
...
+    BOT_1 = 'bot_1'
+    BOT_2 = 'bot_2'
+    BOT_NAME = 'name'
+    BOT_POINTS = 'points'
+    ROUNDS = 'rounds'
+    DATE = 'date'
+
+
+@unique
+class BotMessageField(Enum):
```
Comment (devdo-eu) [inline enumeration.py:56]: `BotMessageFields` here

File: enumeration.py [hunk 3]
```diff
@@ -39,3 +39,42 @@ class RoundAdvantage(IntEnum):
...
+    ROUND = 'round'
+    BOT_1 = 'bot_1'
+    BOT_2 = 'bot_2'
+    ID = 'id'
+    NAME = 'name'
+    POINTS = 'points'
+
+
+@unique
+class BotField(Enum):
```
Comment (devdo-eu) [inline enumeration.py:69]: `BotFields` here

File: enumeration.py [hunk 4]
```diff
@@ -39,3 +39,42 @@ class RoundAdvantage(IntEnum):
...
+    NAME = 'name'
+    USED = 'used'
+    TIME = 'time'
+    POINTS = 'points'
+
+
+@unique
+class RulesFileField(Enum):
```
Comment (devdo-eu) [inline enumeration.py:77]: `RulesFileFileds` here

File: battleground.py [hunk 5]
```diff
@@ -45,38 +45,32 @@ def __init__(self, threadID, name, server, bot_1, bot_2):
...
-            data['GAME_ID'] = self.threadID
-            data['BOT_1']['NAME'] = bot_1.name()
-            data['BOT_1']['POINTS'] = bot_1.points()
-            data['BOT_2']['NAME'] = bot_2.name()
-            data['BOT_2']['POINTS'] = bot_2.points()
-            data['ROUNDS'] = self.passedRounds
-            data['DATE'] = datetime.now().strftime('%Y-%m-%dT%H:%M:%S')
-            game_list['GAMES'].append(data)
+            data[GamesListFileField.GAME_ID.value] = self.threadID
+            data[GamesListFileField.BOT_1.value][GamesListFileField.BOT_NAME.value] = bot_1.name()
```
Comment (devdo-eu) [inline battleground.py]: These lines are too long. Try to keep up to 80 chars in single line of source, no more.
On the other hand: it's just selecting an element from a two-dimensional array, and it already takes a whole line of 80 characters - something is really off with it...

File: battleground.py [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
-import rules
-import json
-import copy
+from collections import defaultdict
 from datetime import datetime
 from pathlib import Path
-from enumeration import RoundWinner, RoundAdvantage
+
+import rules
+from enumeration import RoundWinner, RoundAdvantage, GamesListFileField, BotMessageField
```
Comment (devdo-eu) [inline battleground.py]: Class names are too long or need to be shadowed.
Maybe use:
```
from enumeration import RoundWinner, RoundAdvantage, \
GamesListFileField as gFields, BotMessageField as bmFields
```
Or some better nicknames for these classes.

**Review by devdo-eu — CHANGES_REQUESTED** [review]
Overall nice changes, but...

File: history/games/insight.ipynb [hunk 7]
```diff
@@ -2,7 +2,7 @@
  "cells": [
```
Comment (devdo-eu) [inline history/games/insight.ipynb]: You need to clear cells before commiting this file.
This way we will not upload unnecessary/auto-generated data.

**Review by devdo-eu — CHANGES_REQUESTED** [review]

File: enumeration.py (hunk 1 above)
Comment (ppolakiewicz) [reply enumeration.py:45]: According to this post
https://stackoverflow.com/questions/15755955/naming-of-enums-in-java-singular-or-plural

and I agree with it enums should have singular name.

File: battleground.py [hunk 8]
```diff
@@ -188,13 +190,19 @@ def concludeGame(self):
...
         elif bots[0].points() < bots[1].points():
-            shortcut['ID'], shortcut['NAME'], shortcut['POINTS'] = RoundWinner.BOT_2, bots[1].name(), bots[1].points()
+            shortcut[bmf.ID.value] = rw.BOT_2
+            shortcut[bmf.NAME.value] = bots[1].name()
+            shortcut[bmf.POINTS.value] = bots[1].points()
         else:
-            shortcut['ID'], shortcut['NAME'], shortcut['POINTS'] = RoundWinner.DRAW, '-', 0
+            shortcut[bmf.ID.value] = rw.DRAW
+            shortcut['NAME'] = '-'
+            shortcut['POINTS'] = 0
```
Comment (devdo-eu) [inline battleground.py]: Why are strings left here, and not enumeric values as in other sources?

File: enumeration.py (hunk 1 above)
Comment (devdo-eu) [reply enumeration.py:45]: ok

**Review by devdo-eu — CHANGES_REQUESTED** [review]

**Review by devdo-eu — APPROVED** [review]
Looks good to me :)

**Pull request merged**